tests/upgrade/basic/task.yaml

```
@@ -21,80 +22,120 @@ execute: |
     #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB/snaps.sh"
 
-    echo "Remove snapd and snap-confine"
-    distro_purge_package snapd snap-confine || true
-    case "$SPREAD_SYSTEM" in
-        fedora-*|centos-*)
-            distro_purge_package snapd-selinux || true
-            ;;
-    esac
+    if [[ "$SPREAD_REBOOT" == "0" ]]; then
+
+        echo "Remove snapd and snap-confine"
+        distro_purge_package snapd snap-confine || true
+        case "$SPREAD_SYSTEM" in
+            fedora-*|centos-*)
+                distro_purge_package snapd-selinux || true
+                ;;
+        esac
+
+        echo "Install previous snapd version from the store"
+        distro_install_package snap-confine snapd
+
+        case "$SPREAD_SYSTEM" in
+            arch-*|centos-*)
+                # start the services on distros where that doesn't happen
+                # automnatically
+                systemctl daemon-reload
+                systemctl restart snapd.socket
+                systemctl enable snapd.socket
+                ;;
+        esac
+
+        # need to be seeded to allow snap install
+        if ! snap wait 2>&1|MATCH "unknown command" ; then
+            snap wait system seed.loaded
+        fi
 
-    echo "Install previous snapd version from the store"
-    distro_install_package snap-confine snapd
+        prevsnapdver=$(snap --version|grep "snapd ")
```

Don't we need to quote the assignment value?
```suggestion
        prevsnapdver="$(snap --version|grep "snapd ")"
```

---

Shellcheck says no, but won't hurt.